crates/cargo-contract/src/cmd/schema.rs

```diff
@@ -0,0 +1,180 @@
+    /// Denotes if output should be printed to stdout.
+    #[clap(flatten)]
+    verbosity: VerbosityFlags,
+    /// Output the result in JSON format
+    #[clap(long, conflicts_with = "verbose")]
+    output_json: bool,
+}
+
+impl VerifySchemaCommand {
+    pub fn run(&self) -> Result<SchemaVerificationResult> {
+        let verbosity: Verbosity = TryFrom::<&VerbosityFlags>::try_from(&self.verbosity)?;
+
+        let mut metadata = serde_json::Value::Null;
+        let mut metadata_source = String::new();
+
+        // 1a. Extract given metadata from .contract bundle
+        if let Some(path) = &self.contract_bundle {
+            let file = File::open(path)
+                .context(format!("Failed to open contract bundle {}", path.display()))?;
+
+            let mut contract_metadata: contract_metadata::ContractMetadata =
+                serde_json::from_reader(&file).context(format!(
+                    "Failed to deserialize contract bundle {}",
+                    path.display()
+                ))?;
+            contract_metadata.remove_source_wasm_attribute();
+
+            metadata = serde_json::value::to_value(contract_metadata)?;
+            metadata_source = path.display().to_string();
+        }
+
+        // 1a. Read metadata file
+        if let Some(path) = &self.metadata {
+            let file = File::open(path)
+                .context(format!("Failed to open metadata file {}", path.display()))?;
+
+            let contract_metadata: contract_metadata::ContractMetadata =
+                serde_json::from_reader(&file).context(format!(
+                    "Failed to deserialize metadata file {}",
+                    path.display()
+                ))?;
+
+            metadata = serde_json::value::to_value(contract_metadata)?;
+            metadata_source = path.display().to_string();
+        }
+
+        // 2. Open schema file
+        let path = &self.schema;
+        let file = File::open(path)
+            .context(format!("Failed to open schema file {}", path.display()))?;
+
+        let schema: serde_json::Value = serde_json::from_reader(&file).context(
+            format!("Failed to deserialize schema file {}", path.display()),
+        )?;
+
+        // 3. Build validator
+
+        // We have to use let-else here, otherwise `&schema` is required to be static
+        let Ok(validator) = JSONSchema::compile(&schema) else {
+            anyhow::bail!("Failed to compile schema to validation tree")
+        };
+
+        // 3. Validate and display error if any
+        validator.validate(&metadata).map_err(|errors| {
+            let error_msg = errors.fold(
+                String::from("Error during schema validation:\n"),
+                |acc, e| format!("{}\n{}", acc, e),
+            );
+            anyhow!(error_msg)
+        })?;
+
+        Ok(SchemaVerificationResult {
+            is_verified: true,
+            metadata_source,
+            schema: self.schema.display().to_string(),
```

Comment:
Why are we keeping the schema as a string in the structure? Perhaps we should convert it to a string during struct serialization?

Reply:
This is a path to a schema file